exaudfclient/base/script_options_parser/ctpg/script_option_lines_ctpg.cc

```diff
@@ -0,0 +1,199 @@
+#include "script_option_lines_ctpg.h"
+#include "ctpg.hpp"
+#include <iostream>
+#include <string>
+#include <sstream>
+
+using namespace exaudf_ctpg;
+using namespace exaudf_ctpg::ftors;
+
+
+namespace ExecutionGraph
+{
+
+namespace OptionsLineParser
+{
+
+
+namespace CTPG {
+
+
+struct Option {
+    std::string key;
+    std::string value;
+    source_point start;
+    source_point end;
+};
+
+using options_type = std::vector<Option>;
+
+namespace ParserInternals {
+
+const auto empty_options()
+{
+    options_type ob;
+    return ob;
+}
+
+const auto to_options(Option&& e)
+{
+    options_type ob{e};
+    return ob;
+}
+
+const auto to_option(std::string_view key, std::string value, source_point sp_begin, source_point sp_end)
+{
+    return Option{std::string(key), value, sp_begin, sp_end};
+}
+
+auto&& add_option(Option&& e, options_type&& ob)
+{
+    ob.push_back(std::move(e));
+    return std::move(ob);
+}
+
+
+
+constexpr char alpha_numeric_pattern[] = R"_([0-9a-zA-Z_]+)_";
+constexpr char option_char_pattern[] = R"_([^;])_";
```

Comment:
Might be not the best name

Reply:
Could be that option char should be actually % instead of ;, because in the value only the escaped ; is allowed.

Reply:
In any case this needs explanation

Reply:
It is not ; , yes needs explanation

Reply:
Better name is probably not_semicolon

Reply:
Yep, that is much clearer.

Reply:
Done